docker-compose.yml

```diff
@@ -0,0 +1,30 @@
+version: "3"
+services:
+
+  api:
+    build: ./api
+    restart: always
+    depends_on:
```

Comment:
`depends_on` will just wait for the container to be running, but not for the service in the container to be in a started up state. So if the mongo container is running, it could be that mongo is still starting up in there. 

Best practice would be to make your code (i.e. api and crawler) tolerant to this and ensure they can handle that (they can wait for it to run somehow before they execute stuff interacting with mongo).
Another way would be a wait script. See details here:

https://docs.docker.com/compose/startup-order/

@amcmurray-bw has used some existing wait script using environment variables indicating what to wait for. Maybe @amcmurray-bw you can link that here? But you can also make your applications safe enough that this is not necessary, which would be the best practice as I said:

>The best solution is to perform this check in your application code, both at startup and whenever a connection is lost for any reason. However, if you don’t need this level of resilience, you can work around the problem with a wrapper script:

Also see https://docs.docker.com/compose/compose-file/#depends_on

Reply:
I thought that `depends_on` waits for the services 😞. But the code is already (most probably) tolerant. I tried to run the application (without docker) without mongo running and it tries to connect to the DB in 30 seconds then fails afterwards. Also I ran the application with mongo then stopped mongo and the application tries for another 30 seconds to reconnect.

Reply:
So it's tolerant as long as mongo starts within 30 seconds, if I get you right?

Reply:
Yup. I believe that this can be configured. Should I change it or 30 secs are good? I think if mongo did not start in 30 secs this means that there is an error most probably.

Reply:
Yeah I suppose for your use case that's fine.